chemprop/v2/utils/utils.py

```diff
@@ -26,8 +20,16 @@ def get(cls, name: str | AutoName) -> AutoName:
             )
 
     @classmethod
-    def keys(cls) -> set[str]:
-        return {e.value for e in cls}
+    def keys(cls) -> Iterator[str]:
+        return (e.name for e in cls)
```

Comment:
`Autoname.keys()` returned a dictionary. Now `EnumMapping.keys()` returns a generator. Is there a reason to not have  `EnumMapping.keys()` return a dictionary so that we don't to wrap it in `set()` (line 171 in test_cgr.py for example). 

As a side note, I appreciate that `.keys()` now returns the names and `.values()` returns the values. That was confusing to me before.

Reply:
`Autoname.keys()` returns a set not a dict. (note there is no `:` inside the `{}`.

Reply:
I decided to change to an `Iterator` because I couldn't guarantee parallel ordering between `keys()` and `values()` when returning `set`s. That is, `dict(zip(EnumMapping.keys(), EnumMapping.values()))` was no guaranteed to return a `dict`-like version of the `EnumMapping`

Reply:
That makes sense to me. I think this PR should then add `set()` everywhere an EnumMapping is used to give its keys or values. The only case I found just now is in `common.py` there is `choices=RxnMode.keys()` which should be `choices=set(RxnMode.values())`.

Reply:
`choices` just needs to be any iterable. The new implementation works the same but is marginally better. You don’t need to make the keys into a `set`